builtin: avoid the looseness of sscanf() parsing

tonycoz opened the pull request.

tonycoz committed:
builtin: avoid the looseness of sscanf() parsing

leonerd reviewed (approved):
Code/test changes LGTM

The `g++` smoker looks unhappy but at first glance it really doesn't look like it's anything caused by this change. Might just be a suprious failure in Test2 itself. Perhaps try rerunning it?

leonerd commented:
`g++` happy now on a rerun

tonycoz committed:
builtin bundles: validate end of string correctly.

This went down a little rabbit hole:

- reporting the version with the NUL stopped at the NUL, so try SVf
- reporting with SVf output an invisible character
- reporting with SVf_QUOTEDPREFIX adds quotes, which changed the
  message a little
